app/views/comments/_comment_form.html.erb

```diff
@@ -107,7 +107,7 @@
         <% elsif comment.persisted? %>
           <%= cancel_edit_comment_link(comment) %>
         <% elsif commentable.is_a?(Comment) || commentable.is_a?(CommentDecorator) %>
-          <%= cancel_comment_reply_link(commentable) %>
+          <%= cancel_comment_reply_link(commentable, remote: !on_js_less_comment_form(commentable)) %>
```

Comment:
It looks like this results in redirecting to the work page (assuming the parent comment is on a work) with the comments section collapsed. That does the job, but I'm not sure if it's the intended behaviour. Thoughts @sarken ?

Reply:
Do you mean when replying from the comment section or somewhere else?

I'm looking at the current behavior of the Cancel button on the reply form and what I see is:

* if you're replying to a comment from an email link, e.g., `http://test.archiveofourown.org/comments/5209801?add_comment_reply_id=5209801`, it 
  * redirects to the comment you're replying to without the reply form open if JavaScript is disabled
  * closes the reply form without reloading if JavaScript is enabled
* if you're replying to a comment from the comment section on the work, it
  * redirects to the work with the comment section closed if JavaScript is disabled
  * closes the reply form without reloading if JavaScript is enabled

So if we're talking about from the comment section on the bottom of a work and not a page like the URL I provided, it's not ideal, but it does seem consistent. I guess in an ideal world it would leave the comment section expanded when it redirected to the work, but I'd probably call that out of scope.

Reply:
Well...

While attempting to clean up a seemingly unused method, I accidentally found the source of the bug that I had tried to circumvent by forcing a redirection there. So it's not needed anymore as the form now collapses down properly even if the page started in "focused" mode, and everything is smooth in the best of worlds.

Hooray for serendipity?

Reply:
Hah, I love when deleting code fixes bugs :laughing:

I guess it's less relevant now, but the case I was talking about was a special case of 2b: opening the reply link in a new tab (so ending up on a page like http://localhost:3000/chapters/170?add_comment_reply_id=42&show_comments=true#comment_42) and then clicking "Cancel"

Reply:
That should work now actually. Cf https://github.com/otwcode/otwarchive/pull/4381/files#diff-6adcc86fa92c4daa39cee6c5bbac11ad8cc3a746852f42c98e8a4f1301cb0eb2R233-R246